src/app.ts

```diff
@@ -0,0 +1,131 @@
+import { PassThrough } from "node:stream";
+import express, { Request } from "express";
+import helmet from "helmet";
+import cors from "cors";
+
+import { z, ZodError } from "zod";
+
+import qrcode from "qrcode";
+
+import * as LE from "@pagopa/logger-express";
+import * as L from "@pagopa/logger";
+
+import {
+  BuildLinkPayload,
+  buildLink,
+  appleAppSiteAssociation,
+  assetLinks,
+} from "./applink";
+
+import { getUrlFromUserAgent } from "./redirect";
+
+import { Config } from "./config";
+
+const logError = (err: unknown, req: Request) => {
+  /* c8 ignore start */
+  if (err instanceof ZodError) {
+    err.issues.forEach((issue) => {
+      req.log?.("error", "invalid input", {
+        issue,
+      });
+    });
+  } else if (err instanceof Error) {
+    req.log?.("error", err.message);
+  }
+  /* c8 ignore end */
+};
+
+export const createApp = (
+  c: Readonly<Config>,
+  logger: L.Logger
+): express.Application => {
+  const app = express();
+
+  app.use(helmet());
+  app.disable("x-powered-by");
+  app.use(express.static("public"));
+  app.use(LE.logger(logger));
+  app.use(LE.access());
+
+  app.get("/health", (req, res) => {
+    res.status(204).end();
+  });
+
+  app.get("/.well-known/apple-app-site-association", (_, res) => {
+    if (c.ios) {
+      res.json(appleAppSiteAssociation(`${c.ios.appId}.${c.ios.bundleId}`));
+    } else {
+      res.status(404).end();
+    }
+  });
+
+  app.get("/.well-known/assetlinks.json", (_, res) => {
+    if (c.android) {
+      res.json(
+        assetLinks(c.android.packageName, ...c.android.sha256CertFingerprints)
+      );
+    } else {
+      res.status(404).end();
+    }
+  });
+
+  app.get("/qrcode.png", cors(), async (req, res) => {
+    const Payload = z.intersection(
+      BuildLinkPayload,
+      z.object({
+        width: z.coerce.number().min(100).max(500).default(250),
+        color: z.string().length(9).regex(/^#/).default("#000000ff"),
+      })
```

Comment:
Why? You can override the default value simply using the querystring 

Maybe the default values will never be used (for example our CTA flow uses a Gray QR code with 150x150 size instead of black-250x250)